### DISCUSSION TITLE
Error Message "Attribute ... was changed but it hasn't associated unique key" on multiple result sets

### DISCUSSION THREAD

**Discussion by Aileron79** [dbeaver/dbeaver#20363] [General]
### Description

I get this error message only when editing a result set that is one of many:
**"Attribute address was changed but it hasn't associated unique key"**

When editing and saving the same result when it is the only result set being returned, there are no issues at all.

See steps to reproduce.

### DBeaver Version

Version 23.1.0.202306041918

### Operating System

Windows 10

### Database and driver

Azure SQL
Microsoft JDBC Driver for SQL Server

### Steps to reproduce

I have this table in an Azure SQL database:

```
CREATE TABLE config (
	fk_tenant_id int NOT NULL,
	fk_type_id int NOT NULL,
	enabled bit DEFAULT 1 NOT NULL,
	address varchar(1000) COLLATE SQL_Latin1_General_CP1_CI_AS NOT NULL,
	CONSTRAINT PK_config PRIMARY KEY (fk_tenant_id,fk_type_id)
);
```
Note that the table has a primary key stretched across two fields.

If I run this statement
`SELECT * FROM config WHERE fk_tenant_id = 1234`
I can edit the address field in the result set and save it. To be precise, the statement returns four rows. I do a search and replace to update the address field, all four rows are changed. Hitting the save button will write the changes to database - no problem.

If I run these two lines in a single request, I get two result sets in two tabs in the result window:
```
SELECT * FROM config WHERE fk_tenant_id = 3456
SELECT * FROM config WHERE fk_tenant_id = 5678
```
The first one will return 8 rows, the second 4.
I do the same thing in the second result set: Search and replace on the address field, all four rows are changed. Hitting the save button now results in this error message:

**"Attribute address was changed but it hasn't associated unique key"**

Basically the same operation works if a single result set is returned, but does not work anymore as soon as there are multiple result sets involved. I doubt this has anything to do with a unique key - it's the same table, the same primary key.

**Comment by ShadelessFox** [comment]
Hello @Aileron79,

Are you executing these queries by selecting them and pressing <kbd>CTRL+ENTER</kbd>? If so, can you try pressing <kbd>ALT+X</kbd> instead to see if the issue still occurs?

Thanks.

**Comment by Aileron79** [comment]
Believe it or not, I cannot reproduce the issue. When running two queries, I cannot edit the result set anymore - no matter if I run it by Ctrl+Enter or Alt+x - these buttons are greyed out and upon double click I cannot change the field values:
![image](https://github.com/dbeaver/dbeaver/assets/16278437/5ad1a2e2-3f1e-428a-b907-c46c56d680b0)
However, if I run a single statment, it works:
![image](https://github.com/dbeaver/dbeaver/assets/16278437/0755b416-b883-4684-9826-0e25311054b1)
Not sure if this is another bug or a feature... Restarted dbeaver a few times, and it is the same version. Same tables and database as well.

**Comment by ShadelessFox** [comment]
Try running this following script using <kbd>ALT+X</kbd> (note the trailing semicolons):

```sql
SELECT * FROM config WHERE fk_tenant_id = 3456;
SELECT * FROM config WHERE fk_tenant_id = 5678;
```

**Comment by Aileron79** [comment]
Ok, it seems to work when using semicolons and running the statements at once using Alt+x. When I reported the issue, I surely ran it by Ctrl+Enter as I was not aware of Alt+x. (What's the difference?) However, I still cannot edit the result set when running both at the same time with ctrl+Enter even using semicolons. Not a problem if this is expected behavior, it just does not appear logical to me.

**Comment by ShadelessFox** [comment]
Using <kbd>CTRL+ENTER</kbd>, you send the entire selection to the database to process. Sometimes, it (the driver or the database itself) might not like that it consists of several queries and respond with invalid metadata that DBeaver can't properly use.

On the other hand, <kbd>ALT+X</kbd> splits queries in the selection by either a line separator or `;` and sends each query to the database. This is a more versatile way to execute multiple queries at once.

**Comment by Aileron79** [comment]
Thanks for the explanation!
